### PR TITLE
Boundary conditions

### PR DESCRIPTION
**Pull request by sammorrell**
Implemented boundary conditions. Includes structures, supporting code and basic integration into MCRT code. Also includes some unit tests.

### CODE REVIEW

File: src/bin/mcrt.rs [hunk 1]
```diff
@@ -84,6 +84,14 @@ fn main() {
         .expect("Failed to link attribute to surfaces.");
     report!(surfs, "surfaces");
 
+    /*
+     * Create a boundary for the simulation with boundary coneditions.
```
Comment (EdHone) [inline src/bin/mcrt.rs:88]: *conditions typo

File: src/geom/domain/boundary.rs [hunk 2]
```diff
@@ -0,0 +1,584 @@
...
+        Self {
+            bounding_box,
+            top: BoundaryCondition::Periodic(padding),
+            bottom: BoundaryCondition::Periodic(padding),
+            north: BoundaryCondition::Periodic(padding),
+            east: BoundaryCondition::Periodic(padding),
+            south: BoundaryCondition::Periodic(padding),
+            west: BoundaryCondition::Periodic(padding),
+        }
+    }
```
Comment (EdHone) [inline src/geom/domain/boundary.rs:75]: Don't necessarily mind this - i quite like how explicit the interface is for creating boundaries, It feels like code itself can probably be trimmed down but at the moment definitely keep it how it is :)

File: src/geom/plane.rs
Comment (EdHone) [inline src/geom/plane.rs:1]: A test for this function would be nice

**Review by EdHone — CHANGES_REQUESTED** [review]
Some minor comments but otherwise this is a great change - the level of integration with the existing systems is great to see

File: src/bin/mcrt.rs (hunk 1 above)
Comment (sammorrell) [reply src/bin/mcrt.rs]: This has been fixed in https://github.com/aetherus-wg/Aetherus/pull/56/commits/59c4f43ab36666c64d2eda2ad01ed217b2f07ea4.

File: src/geom/domain/boundary.rs (hunk 2 above)
Comment (sammorrell) [reply src/geom/domain/boundary.rs:75]: Agreed! It's quite verbose, but it is intended as a first working version that provides the flexibility for the MPI implementation to follow. More than happy to revisit this in quality-of-life improvements in the future.

**Review by EdHone — APPROVED** [review]

**Comment by EdHone** [comment]
Thanks for humoring my nitpicks! Merge away, Sam!
